marshmallow/fields.py

```diff
@@ -1016,32 +1027,86 @@ def _deserialize(self, value, attr, data):
 class Date(Field):
     """ISO8601-formatted date string.
 
+    :param strict: If True, only accept Date objets. If False, will accept any
+        object that has integer ``year``, ``month`` and ``day`` values.
+        Default False.
+    :param format: Either ``"iso"`` (for ISO8601) or a date format string.
+        If `None`, defaults to "iso".
     :param kwargs: The same keyword arguments that :class:`Field` receives.
     """
     default_error_messages = {
         'invalid': 'Not a valid date.',
         'format': '"{input}" cannot be formatted as a date.',
     }
 
-    def _serialize(self, value, attr, obj):
-        if value is None:
-            return None
+    DATEFORMAT_SERIALIZATION_FUNCS = {
+        'iso': utils.to_iso_date,
+        'iso8601': utils.to_iso_date,
+    }
+
+    DATEFORMAT_DESERIALIZATION_FUNCS = {
+        'iso': utils.from_iso_date,
+        'iso8601': utils.from_iso_date,
+    }
+
+    DEFAULT_FORMAT = 'iso'
+
+    def __init__(self, strict=False, format=None, **kwargs):
+        super(Date, self).__init__(**kwargs)
+        self.strict = strict
+        self.dateformat = format
+
+    def _add_to_schema(self, field_name, schema):
+        super(Date, self)._add_to_schema(field_name, schema)
+        self.dateformat = self.dateformat or \
+            schema.opts.dateformat or \
+            self.DEFAULT_FORMAT
+
+    def _validated(self, value):
+        if self.strict and type(value) != dt.date:
+            self.fail('format', input=value)
         try:
-            return value.isoformat()
-        except AttributeError:
+            return dt.date(value.year, value.month, value.day)
+        except (TypeError, AttributeError, ValueError):
             self.fail('format', input=value)
-        return value
+
+    def _serialize(self, value, attr, obj):
```

Comment:
A lot of this new code seems to be duplicated from `DateTime`. Can we use inheritance to avoid duplicating this code?

Reply:
The only major impediment is the distinction between `dateformat` and `datetimeformat` in Meta, but I think I can write a function to change that between the two classes.

How are you envisioning the class inheritance graph for this? `class Date(DateTime)` or `class Date(FormattedField)` and `class DateTime(FormattedField)` or something similar?

Reply:
I don't think python has a defined hierarchy between the two modules. I think it would be valid to treat `Date` as a specialized form of `DateTime`. `class DateTime(Field)` and `class Date(DateTime)` would probably generate the smallest diff also.

> I can write a function to change [the meta format] between the two classes

:+1: Good catch.